api/src/main/java/io/strimzi/api/kafka/model/KafkaResources.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+public class KafkaResources {
+    private KafkaResources() { }
+
+    /**
+     * Returns the name of the Zookeeper {@code StatefulSet} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Zookeeper {@code StatefulSet}.
+     */
+    public static String zookeeperStatefulSetName(String clusterName) {
+        return clusterName + "-zookeeper";
+    }
+
+    /**
+     * Returns the name of the Kafka {@code StatefulSet} for a {@code Kafka} cluster of the given name.
+     * @param clusterName  The {@code metadata.name} of the {@code Kafka} resource.
+     * @return The name of the corresponding Kafka {@code StatefulSet}.
+     */
+    public static String kafkaStatefulSetName(String clusterName) {
+        return clusterName + "-kafka";
```

Comment:
I'm not sure I understand how does this class fit.

a) I'm not sure it belongs to the `api` module. But I'm quite sure you can convince me about it ;-).
b) This seems to be covered by other methods in `cluster-operator` module (such as `kafkaClusterName(...)` in `KafkaCluster`). But I do not see this PR to change the other parts. So it seems like it created a duplication of code which have to be maintained on two places now.

Reply:
I think this class does belong in the API module because users of the API will quite likely be interested in monitoring the readiness of the StatefulSet, which is deployed by their custom resource. If they do that they will end up hard-coding the naming scheme we have into their application anyway. It's better to encapsulate these naming schemes into a class which we control. That way the user doesn't have to care so much about the details and should we ever change it the user will be more isolated from that change.

You're completely correct that this duplicates functions we have elsewhere. TBH this is a change I've been wanting to make for a while, and I felt it was easier to just start on it now and incrementally refactor the rest of the codebase over time to use these new functions than it was to embark on a refactoring PR.

Reply:
Ok, I can understand the motivation. But I think you should just remove it from this PR and open a separate PR to move all these functions to the `api` module. You should not start this and keep it in two places as duplicates. Or at least as a bare minimum if you really insist on doing this in this PR, you should move the affected methods to avoid duplicates.